### PR TITLE
Cloudstack 8612

### PR DESCRIPTION
**Pull request by likitha**
To ensure that setting a high timeout value for snapshot backup operation will not affect vCenter connections made for any other operations besides volume snapshot, split the existing vCenter session timeout configuration into two configurations,
1. To tune timeout value for CS's session vCenter during snapshot backup operation - 'vmware.snapshot.backup.session.timeout' (new; default: 20 minutes)
2. For all other operations - 'vmware.vcenter.session.timeout' (existing; default: 20 minutes)

Additional change -
Users could potentially run into a timeout issue while trying to snapshot large VMs. There is a timeout for async-jobs that ensures that any job that has been in process for too long is cancelled by CS MS. This timeout is configurable using a global configuration ‘job.cancel.threshold.minutes’ and it defaults to 1 hour.
Now for large volumes even though users configure other snapshot timeout values (‘vmware.snapshot.backup.session.timeout’ and ‘backup.snapshot.wait‘) to a very high value, the snapshot operation will still fail because the job will be cancelled in 1 hour.
Hence proposing a configuration that will allow admins to separate out the job cancellation timeout for ‘volume snapshot’ jobs from other vm work jobs - 'volume.snapshot.job.cancel.threshold'.

### CODE REVIEW

File: framework/jobs/src/org/apache/cloudstack/framework/jobs/dao/SyncQueueItemDaoImpl.java [hunk 1]
```diff
@@ -141,23 +141,46 @@ public int getActiveQueueItemCount(long queueId) {
...
-        if(exclusive)
-            return lockRows(sc, null, true);
-        return listBy(sc, null);
+        String sql = "SELECT s.id, s.queue_id, s.content_type, s.content_id, s.queue_proc_msid, s.queue_proc_number, s.queue_proc_time, s.created FROM sync_queue_item AS s " +
+                "JOIN async_job as a ON s.content_id = a.id WHERE s.queue_proc_msid IS NOT NULL AND s.queue_proc_number IS NOT NULL AND s.queue_proc_time IS NOT NULL" +
+                " AND ((a.job_cmd NOT LIKE ? AND s.queue_proc_time < ? ) OR (a.job_cmd LIKE ? AND s.queue_proc_time < ?))";
+        TransactionLegacy txn = TransactionLegacy.currentTxn();
+        PreparedStatement pstmt = null;
+        try {
+            pstmt = txn.prepareAutoCloseStatement(sql);
```
Comment (DaanHoogland) [inline framework/jobs/src/org/apache/cloudstack/framework/jobs/dao/SyncQueueItemDaoImpl.java]: please use try-with-resource

File: framework/jobs/src/org/apache/cloudstack/framework/jobs/dao/SyncQueueItemDaoImpl.java [hunk 2]
```diff
@@ -141,23 +141,46 @@ public int getActiveQueueItemCount(long queueId) {
...
+            pstmt.setString(1, "%" + jobCmd + "%");
+            pstmt.setString(2, dateString1);
+            pstmt.setString(3, "%" + jobCmd + "%");
+            pstmt.setString(4, dateString2);
+            ResultSet rs = pstmt.executeQuery();
```
Comment (DaanHoogland) [inline framework/jobs/src/org/apache/cloudstack/framework/jobs/dao/SyncQueueItemDaoImpl.java]: please use try-with-resource

File: plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 3]
```diff
@@ -5035,9 +5039,16 @@ public VmwareHypervisorHost getHyperHost(VmwareContext context) {
     @Override
     public VmwareContext getServiceContext(Command cmd) {
         VmwareContext context = null;
+        int vCenterSessionTimeout;
+        if (cmd != null && cmd.getContextParam("vCenterSessionTimeout") != null) {
+            vCenterSessionTimeout = NumbersUtil.parseInt(cmd.getContextParam("vCenterSessionTimeout"), 1200000);
```
Comment (DaanHoogland) [inline plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java]: why not use _vCenterSessionTimeout here?

**Comment by DaanHoogland** [comment]
Can you add unit- and/or integration tests?

**Comment by remibergsma** [comment]
Who wants to step in and finish this work? It seems the original author is not able to finish it. If no one steps in, we'll have to close the PR without merging it so please help :-).

**Comment by rhtyd** [comment]
tag:vmware-pickup

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 136
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by sureshanaparti** [comment]
Continuing with the code changes here in a new PR #2099.

**Comment by DaanHoogland** [comment]
closing as @sureshanaparti handles this in #2099

**Pull request closed without merging**